app/src/main/java/org/mifos/mobile/ui/registration/RegistrationViewModel.kt

```diff
@@ -49,6 +57,126 @@ class RegistrationViewModel @Inject constructor(private val userAuthRepositoryIm
         return !PatternsCompat.EMAIL_ADDRESS.matcher(emailText.trim()).matches()
     }
 
+    fun areFieldsValidated(
+        context: Context,
```

Comment:
dont use context in a viewmodel. could lead to memory leaks + ViewModels are intended to be lifecycle-aware and should not depend on any UI-related components or contexts.
Do these validations in the compose screen itself